forest/src/cli/auth_cmd.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::stringify_rpc_err;
+use rpc_client::{auth_new, new_client};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub enum AuthCommands {
+    /// Create a new Authentication token with given permission
+    #[structopt(about = "<String> Create Authentication token with given permission")]
+    CreateToken {
+        #[structopt(
+            short,
+            help = "permission to assign to the token, one of: read, write, sign, admin"
+        )]
+        perm: String,
+    },
+}
+
+impl AuthCommands {
+    pub async fn run(&self) {
+        // TODO handle cli config
+        match self {
+            Self::CreateToken { perm } => {
+                let perm: String = perm.parse().unwrap();
+                let mut client = new_client();
+
+                let obj = auth_new(&mut client, perm)
+                    .await
+                    .map_err(stringify_rpc_err)
```

Comment:
If you arent going to use the error message, its probably better to remove the mapping

Reply:
The error message is used, the create_token method returns a JWTResult which is then converted to a JsonRpcError in the event of Err. The Error message is used in the unwrap which shows up in the client CLI upon an error

Reply:
Ahhh right because when it crashes the user will see the error, I see your point